Search/interpolation_search.cpp

```diff
@@ -1,40 +1,37 @@
 #include<iostream>
 using namespace std;
 
-//function to search the value in an array using interpolation search
+// function to search the value in an array using interpolation search
 int search(int arr[], int value, int len) {
         int low = 0, high, mid;
         high = len-1;
-        while(arr[low] <= value && arr[high] >= value) {
-                        mid = (low + (( value-arr[low])*(high-low)) / (arr[high]-arr[low]));
-                        if(arr[mid] > value)
-                                high = mid-1;
-                        else if (arr[mid] < value)
-                                low = mid+1;
-                        else
-                                return mid; 
+        while (arr[low] <= value && arr[high] >= value) {
+            mid = (low + ((value-arr[low])*(high-low)) / (arr[high]-arr[low]));
+            if (arr[mid] > value)
+                     high = mid-1;
+            else if (arr[mid] < value)
+                     low = mid+1;
+                 else
+                     return mid; 
          }
          if(arr[low] == value)
                 return low;
          return 0;
 }
 
 int main() {
-        int n,value,array[100],re;
-        cout<<"Enter the size of array(less than 100) : ";
-        cin>>n;
-        cout<<"Enter the values of array in ascending (increasing) order : "<<endl;
-	
-        for(int i=0;i<n;i++)
-                cin>>array[i];
-
-        cout<<"Enter the value you want to search : ";
-        cin>>value;
-        re=search(array,value,n);
-
-        if(re==0)
-                cout<<"Entered value is not in the array"<<endl;
+        int n ,value ,array[100] ,re;
+        cout << "Enter the size of array(less than 100) : ";
+        cin >> n;
+        cout << "array in ascending (increasing) order : " << endl;
```

Comment:
Need to add __std::__ like this:
```python
         std::cout << "Enter the size of array(less than 100) : ";
         std::cin >> n;
         std::cout << "array in ascending (increasing) order : " << std::endl;
```

Reply:
Ohk but why is it necessary

Reply:
https://stackoverflow.com/questions/1265039/using-std-namespace
TL;DR it can eliminate hard to find bugs where someone defines xyz that there is also a std::xyz